halo-dev/halo
# Add user support for authentication and authorization

#### What type of PR is this?

/kind feature
/kind api-change

#### What this PR does / why we need it:

This PR aims to provide user support for authentication and authorization and prepares to provide ability for personal access token.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

##### How to test?

1. Start the application
2. See the latest log and find the password of user admin
3. Try to access <http://localhost:8090/> in the browser and you will be redirected to login page
4. Login with `admin` and `random_password_in_log`
5. Try to access:
    - <http://localhost:8090/api/v1alpha1/users>
    - <http://localhost:8090/api/v1alpha1/roles>
    - <http://localhost:8090/api/v1alpha1/rolebindings>

#### Does this PR introduce a user-facing change?

```release-note
None
```
